include/envoy/network/connection_handler.h

```diff
@@ -43,13 +43,29 @@ class ConnectionHandler {
    */
   virtual void addListener(ListenerConfig& config) PURE;
 
+  /**
+   * Adds a listener to the handler and replace the existing listener.
+   * @param overrided_listener supplies the tag of the existing listener.
+   * @param config listener configuration options.
+   */
+  virtual void addIntelligentListener(uint64_t overrided_listener, ListenerConfig& config) PURE;
+
   /**
    * Remove listeners using the listener tag as a key. All connections owned by the removed
    * listeners will be closed.
    * @param listener_tag supplies the tag passed to addListener().
    */
   virtual void removeListeners(uint64_t listener_tag) PURE;
 
+  /**
+   * Remove the untracked filter chains and the connections in the listener. All connections owned
+   * by the removed listeners will be closed. Once all the connections are destroyed(connections
```

Comment:
What is meant by an "untracked filter chain"? Can you clarify what this means and who is going to call this function and when?

Reply:
As a quick response: active listener at worker thread tracks 2 groups of connections, 1 group is owned by the active_filter_chains, another group is owned by the filter chains under drainage.
Upon drain timeout, the latter group of connections need to be force closed.

The expected flow is triggered by drain timeout, starting from `void ListenerManagerImpl::drainFilterChains(ListenerImplPtr&& old_listener, ListenerImpl& new_listener)`

Reply:
BTW: the untracked filter chains are referring to the read filter chains in design doc